### PR TITLE
feat: add OctoApp support

### PR DESCRIPTION
**Pull request by crysxd**
This PR adds support for OctoApp. I modified:

- Main menu to reflect OctoApp
- Update menu to reflect OctoApp
- Remove menu to reflect OctoApp
- Redme

My scripts are copied and modified from the OctoEverywhere ones, as my plugin is a fork of OctoEverywhere

### CODE REVIEW

**Comment by dw-0** [comment]
Hi! Thank you for your contribution.
As you might have seen (or maybe not), im currently in the process of a full KIAUH rewrite. That means migrating the whole script to Python 3. At the same time there will be some fundamental structural changes. This includes how i consider "non core components" to be implemented in KIAUH. There was a feature request for adding your plugin to KIAUH which you can see here: https://github.com/dw-0/kiauh/issues/408

I mentioned that i rather see this integrated as a, what i will probably call "Community Extension", for KIAUH.
An idea and poc of how this would look like can be seen here: https://github.com/dw-0/kiauh/issues/330
In the v6 dev branch there are already 2 extensions implemented, so if you're interested about the "look and feel" feel free to check it out there as well. As i notices, some extensions might require not only install and remove, but also update. So i guess i will add an optional update entry point to the module.

Long story short:
Im hesitant in merging this PR, as it will also require a rewrite into becoming a KIAUH community extension soon. And it might be too early to also start migrating this PR to a community extension for v6 already. Although the extension feature is already implemented in v6 and looks fine to me, i would still consider it WIP. So just to let you know, this PR has to stay on hold for a while.

**Comment by crysxd** [comment]
I understand that the timing of this request is not ideal and I also took my sweet time to work on this.

I'm also not in an immense rush, so if I get an idea on what "soon" means for the community extensions I'm happy to wait and rework my integration. I just don't want to get in a situation where this now doesn't get merged and the community extensions take another 6 months to make their way into the production version of KIAUH as I get a fair amount of requests on adding OctoApp to KIAUH.

As my modifications here are a 1:1 copy of the OctoEverywhere integration, there is also not a lot of overhead on migrating my changes to any new system as the OE changes would need to get the same treatment already. 

My proposal would be: If the community extensions are right around the corner I'm happy to wait and then integrate via this path. But if the community extensions are still more than 1-2 months out, I'd prefer to get this merged now and I'm then happy to do the extra work required to migrate them over. If you consider the v6 somewhat stable I can do so already now.

**Comment by dw-0** [comment]
> My proposal would be: If the community extensions are right around the corner I'm happy to wait and then integrate via this path. But if the community extensions are still more than 1-2 months out, I'd prefer to get this merged now and I'm then happy to do the extra work required to migrate them over. If you consider the v6 somewhat stable I can do so already now.

I think that would be fine with me. I know you put in the work and holding this PR until i am finished with v6 (earlier those extensions won't make it to production) wouldn't be ideal. I will have a look at this PR later today and review it. And as soon as v6 comes live (it is not stable right now), that feature can get migrated to a community extension.

File: README.md [hunk 1]
```diff
@@ -154,18 +154,20 @@ prompt and confirm by hitting ENTER.
...
 <th><h3><a href="https://octoeverywhere.com/?source=kiauh_readme">OctoEverywhere For Klipper</a></h3></th>
+<th><h3><a href="https://github.com/crysxd/OctoPrint-OctoApp">OctoApp For Klipper</a></h3></th>
 <th><h3></h3></th>
 </tr>
 
 <tr>
 <th><a href="https://github.com/Clon1998/mobileraker_companion"><img src="https://raw.githubusercontent.com/Clon1998/mobileraker/master/assets/icon/mr_appicon.png" alt="OctoEverywhere Logo" height="64"></th>
 <th><a href="https://octoeverywhere.com/?source=kiauh_readme"><img src="https://octoeverywhere.com/img/logo.svg" alt="OctoEverywhere Logo" height="64"></a></th>
-<th></th>
+<th><a href="https://octoeverywhere.com/?source=kiauh_readme"><img src="https://firebasestorage.googleapis.com/v0/b/octoapp-4e438.appspot.com/o/resources%2Foctoeverywhere%2FA.svg?alt=media&token=b5c9ac00-f6ca-484b-8dbf-34db944b8e0a" alt="OctoApp Logo" height="64"></a></th>
```
Comment (dw-0) [inline README.md:164]: You might want to point the URL in the a-tag to the correct ressource.

File: scripts/octoapp.sh [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+      blank_line
+      echo -e "| The setup will apply the same names to                |"
+      echo -e "| OctoApp                                        |"
+      blank_line
+      echo -e "| Please select the number of OctoApp instances  |"
+      echo -e "| to install. Usually one OctoApp instance per   |"
+      echo -e "| Moonraker instance is required, but you may not       |"
+      echo -e "| install more OctoApp instances than available  |"
+      echo -e "| Moonraker instances.                                  |"
+      bottom_border
```
Comment (dw-0) [inline scripts/octoapp.sh:84]:
```suggestion
      blank_line
      echo -e "| The setup will apply the same names to OctoApp        |"
      blank_line
      echo -e "| Please select the number of OctoApp instances to      |"
      echo -e "| install. Usually one OctoApp instance per Moonraker   |"
      echo -e "| instance is required, but you may not install more    |"
      echo -e "| OctoApp instances than available Moonraker instances. |"
      bottom_border
```

File: scripts/octoapp.sh [hunk 3]
```diff
@@ -0,0 +1,385 @@
...
+  status_msg "Cloning OctoApp..."
+  ### force remove existing repos
+  [[ -d "${OCTOAPP_DIR}" ]] && rm -rf "${OCTOAPP_DIR}"
+
+  cd "${HOME}" || exit 1
+  if ! git clone "${repo}" "${OCTOAPP_DIR}"; then
+    print_error "Cloning OctoApp from\n ${repo}\n failed!"
+    exit 1
+  fi
+}
```
Comment (dw-0) [inline scripts/octoapp.sh:174]: This function is declared twice in this file. This is coming form the OctoEverywhere script, where this function is also declared twice. We can simply remove that one here.

**Review by dw-0 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (crysxd) [reply README.md:164]: Thank you, too much copy and paste :D

File: scripts/octoapp.sh (hunk 3 above)
Comment (crysxd) [reply scripts/octoapp.sh:172]: Removed. I ping Quinn about the duplicate function in OE, but as both are identical there is not much harm done

**Comment by crysxd** [comment]
I completed all requested changes, thank you for checking this so quickly!
There is also one small change, I now call the install entry "OctoApp for Klipper" to make it more clear that this is indeed a Klipper addon and not for OctoPrint. I did not change the other menus as the text was too long. If you want to I can revert the change or copy it to the other menus, making them a tad wider

**Review by dw-0 — APPROVED** [review]

**Comment by crysxd** [comment]
Thank you! :)

**Comment by dw-0** [comment]
Thank you!

> I did not change the other menus as the text was too long. If you want to I can revert the change or copy it to the other menus, making them a tad wider

I think it is fine for now. We can leave it as is and will tackle that when v6 is around the corner :)

**Comment by dw-0** [comment]
Hey @crysxd  , are you still interested in adding the python version of the OctoApp installer to KIAUH?
It should reside as a community extension in the project. Feel free to contact me either here or on the Klipper Discord (th33xitus) and we can discuss any details or in case you have questions on how to proceed.
